pkg/status/status.go

```diff
@@ -225,14 +186,6 @@ func (r *reporter) syncChannelReceiver() {
 		select {
 		case <-r.stopCh:
 			return
-		case err := <-r.syncCh:
-			if err == nil {
-				r.syncRatio.ReportSyncEvent()
-			} else {
-				r.syncRatio.ReportFailedSync()
-			}
-			failedSyncs, syncs := r.syncRatio.GetSyncs()
```

Comment:
It seems like we could remove some of the other methods that existed when this was more complicated, such as `ensureClusterOperator` and `setStatusCondition`, etc...

Reply:
I think we should write one function per task even if the system isn't very complicated. I am all for code readability whether the system is simple or complex

Reply:
Right - but I feel like a lot of the complexity in this code is still present in the methods that were left behind. The methods themselves should be simplified.

Reply:
This work can be done as part of [this bug](https://bugzilla.redhat.com/show_bug.cgi?id=1857680)